Sources/SmithyXML/Reader/Reader.swift

```diff
@@ -0,0 +1,317 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import typealias SmithyReadWrite.ReadingClosure
+import struct Foundation.Date
+import struct Foundation.Data
+import enum SmithyTimestamps.TimestampFormat
+import struct SmithyTimestamps.TimestampFormatter
+
+public class Reader {
```

Comment:
`Reader` is the type that is used to deserialize XML data, similar to `Writer` for XML serialization.

A tree of `Reader`s is constructed that matches the tree structure of the XML.  Individual `Reader`s are used to access the data stored within individual XML elements.